lib/CMakeLists.txt

```diff
@@ -0,0 +1,31 @@
+### lib/CMakeLists.txt ###
+#
+#
+
+## Platform and system libraries
+include(platform_libs.cmake)
+
+## imgui:
+execute_process(
+	COMMAND git submodule update --init imgui
+	WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
+)
+
+add_library(imgui imgui/imgui.cpp imgui/imgui_draw.cpp)
+target_include_directories(imgui PUBLIC "${CMAKE_CURRENT_SOURCE_DIR}/imgui")
+target_link_libraries(imgui ${PLATFORM_LIBS})
+target_compile_definitions(imgui PUBLIC "-DImDrawIdx=unsigned int") # short is not enough for us (long causes core dumps)
+set(LIBS ${LIBS} imgui)
+
```

Comment:
Yeah sorry, I did miss that.
> without having to edit the main target whenever we add or drop a dependency
In fact I prefer this. But the way you do it is perfectly ok.